query-engine/query-engine-node-api/src/tracer.rs

```diff
@@ -0,0 +1,87 @@
+use async_trait::async_trait;
+use napi::threadsafe_function::{ThreadsafeFunction, ThreadsafeFunctionCallMode};
+use opentelemetry::{
+    global, sdk,
+    sdk::{
+        export::trace::{ExportResult, SpanData, SpanExporter},
+        propagation::TraceContextPropagator,
+    },
+    trace::TracerProvider,
+};
+use std::fmt::{self, Debug};
+
+/// Pipeline builder
+#[derive(Debug)]
+pub struct PipelineBuilder {
+    trace_config: Option<sdk::trace::Config>,
+}
+
+/// Create a new stdout exporter pipeline builder.
+pub fn new_pipeline() -> PipelineBuilder {
+    PipelineBuilder::default()
+}
+
+impl Default for PipelineBuilder {
+    /// Return the default pipeline builder.
+    fn default() -> Self {
+        Self { trace_config: None }
+    }
+}
+
+impl PipelineBuilder {
+    /// Assign the SDK trace configuration.
+    #[allow(dead_code)]
+    pub fn with_trace_config(mut self, config: sdk::trace::Config) -> Self {
+        self.trace_config = Some(config);
+        self
+    }
+}
+
+impl PipelineBuilder {
+    pub fn install_simple(mut self, log_callback: ThreadsafeFunction<String>) -> sdk::trace::Tracer {
+        global::set_text_map_propagator(TraceContextPropagator::new());
+        let exporter = ClientSpanExporter::new(log_callback);
+
+        let mut provider_builder = sdk::trace::TracerProvider::builder().with_simple_exporter(exporter);
+        // This doesn't work at the moment because we create the logger outside of an async runtime
+        // we could later move the creation of logger into the `connect` function
+        // let mut provider_builder = sdk::trace::TracerProvider::builder().with_batch_exporter(exporter, runtime::Tokio);
+        // remember to add features = ["rt-tokio"] to the cargo.toml
+        if let Some(config) = self.trace_config.take() {
+            provider_builder = provider_builder.with_config(config);
+        }
+        let provider = provider_builder.build();
+        let tracer = provider.tracer("opentelemetry");
+        let _ = global::set_tracer_provider(provider);
```

Comment:
is this `let _` necessary? From experience, completely ignoring the return value no matter what has lead to bugs — if it's a result, calling `.ok()` on it will achieve the same thing, but it will break if the return type changes (to a future, for example), which I find good.

Reply:
Yep, I'd just panic here if the tracer provider cannot be set.

Reply:
This just returns the GlobalTracer. So I've just removed the `let _`